File: CourseApp.API/Model/Exam .cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CourseApp.API.Model
{
    public class Exam
    {
        public int Id { get; set; }
        public int AuthorId { get; set; }
        public User Author { get; set; }
        public string Name { get; set; }
        public DateTime DatePublished { get; set; }
        public string Description { get; set; }
        public ICollection<Question> Questions { get; set; }
        public TimeSpan? Duration { get; set; }
        public bool? Finished { get; set; }
        [Required]
        public byte[] PasswordHash { get; set; }
        [Required]
        public byte[] PasswordSalt { get; set; }


    }
}

File: CourseApp-SPA/src/app/_models/exam.ts
import { Question } from './question';
import { User } from './user';

export interface Exam {
  id: number;
  authorKnownAs: string;
  name: string;
  description: string;
  datePublished: Date;
  duration?: string;
  questions?: Question[];
  delete: boolean;
}
export interface ExamForCreation extends Exam {
  password: string;
}
